exporter/exporterhelper/internal/queuebatch/config.go

```diff
@@ -0,0 +1,100 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package queuebatch // import "go.opentelemetry.io/collector/exporter/exporterhelper/internal/queuebatch"
+
+import (
+	"errors"
+	"time"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/exporter/exporterbatcher"
+)
+
+// Config defines configuration for queueing and batching incoming requests.
+type Config struct {
+	// Enabled indicates whether to not enqueue and batch before exporting.
+	// TODO: This will be removed when Optional is available.
```

Comment:
How `Optional` can replace this if we keep `Enabled: true` by default?

Reply:
I was thinking that users will embed this in the config as optional. For example in otlpexporter we will have `Optional[queuebatch.Config] sending_queue`, the problem with that is there is no way to disable if otlpexporter enables it by default.

Reply:
I was wondering the same. In a parallel PR #12652 you are replacing the disabled behavior -- does that PR reflect how you will change the "disabled" case after Optional is available, in other words? Or, does disabling the queue/batch mean "synchronously call the exporter"?

Reply:
> I want to get to "disabling" means equivalent of the whole component not being present at all.

How disabling would look from user's perspective? By setting `sending_queue` to `nil`?

Reply:
Not sure yet how the design of Optional will be, and if it supports opt-out or only opt-in. So, just to avoid more comments/questions will remove the TODO, but will keep this in my mind if there is an opportunity to use Optional for this.